Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/InternalExchangeManager.java

```diff
@@ -161,12 +162,44 @@ public String getEndpointURL(String hcid, String sServiceName,
 
     @Override
     protected String getAPI_SPEC(ADAPTER_API_LEVEL spec_level
-    ) {
+        ) {
         return spec_level.toString();
     }
 
     @Override
     protected ADAPTER_API_LEVEL getAPI_SPEC_ENUM(String version) {
         return NhincConstants.ADAPTER_API_LEVEL.valueOf(version);
     }
+
+    public static boolean updateServiceUrl(String serviceName, String url) throws Exception {
+        String sHomeCommunityId = HELPER.getHomeCommunityFromPropFile();
+
+        OrganizationType newOrganization = getInstance().getOrganization(sHomeCommunityId);
+        if (null == newOrganization) {
+            return false;
+        }
+
+        EndpointConfigurationType endpointUrl = HELPER.getEndPointConfigBasedOnSpecVersion(
+            HELPER.getServiceEndpointType(newOrganization, serviceName),
+            NhincConstants.ADAPTER_API_LEVEL.LEVEL_a0.name());
+        endpointUrl.setUrl(url);
+
+        ExchangeInfoType exchangeInfo = getExchangeInfo();
+        List<OrganizationType> organizations = HELPER.getOrganizationTypeBy(exchangeInfo);
```

Comment:
Can you use List<OrganizationType> getAllOrganizations() method from AbstractExchangeManager here?